### PR TITLE
Initial support for Blob binding

### PR DESCRIPTION
**Pull request by elprans**
Input bindings are always passed as an `InputStream` object which
implements `BufferedIOBase` and is a read-only non-seekable stream.

Output parameters can be strings, bytes, or file-like objects.

### CODE REVIEW

File: azure/worker/type_converters.py [hunk 1]
```diff
@@ -107,3 +107,46 @@ def from_proto(cls, data: protos.TypedData) -> typing.Any:
...
+
+class BlobConverter(type_meta.InConverter,
+                    type_meta.OutConverter,
+                    binding=type_meta.Binding.blob):
+
+    @classmethod
+    def check_python_type(cls, pytype: type) -> bool:
+        return (issubclass(pytype, (str, bytes, bytearray,
+                                    azf.InputStream) or
+                hasattr(pytype, 'read')))
```
Comment (1st1) [inline azure/worker/type_converters.py]: Also check if `pytype.read` is a callable? Why is duck typing a good idea here?

File: azure/functions/_abc.py [hunk 2]
```diff
@@ -88,3 +89,14 @@ class TimerRequest(abc.ABC):
...
+
+
+class InputStream(io.BufferedIOBase, abc.ABC):
+
+    @abc.abstractmethod
+    def read(self, size=-1) -> bytes:
+        pass
+
+    @abc.abstractmethod
+    def read1(self, size=-1) -> bytes:
```
Comment (1st1) [inline azure/functions/_abc.py]: Why do we need `read1`?

File: azure/worker/type_converters.py (hunk 1 above)
Comment (elprans) [reply azure/worker/type_converters.py]: Duck typing is to support `file-like`.  There's no standard ABC for that, but I can add one if you think that would be cleaner.

File: azure/functions/_abc.py (hunk 2 above)
Comment (elprans) [reply azure/functions/_abc.py]: I was under the false impression that `BufferedIOBase` required `read1` to be implemented.  Removed.

**Pull request merged**